include/private/tdydiscretizationimpl.h

```diff
@@ -0,0 +1,32 @@
+#if !defined(TDYDISCRETIZATION_H)
+#define TDYDISCRETIZATION_H
+
+#include <petsc.h>
+#include <private/tdydmimpl.h>
+
+
+typedef struct {
+
+  TDyDM *tdydm;
+  TDyUGrid *ugrid;
```

Comment:
I remember you talking a bit about the DM "shell" type, and that you were having some issues using it. Can you remind me of the issues you encountered? It seems like it would be more convenient to create a DM shell for the PFlotran-style  FV mesh and use the "ugrid" type as its "context". Then we wouldn't have to carry around this additional `TDyDiscretization` type--we could just check whether the DM is a shell, and if it is, we know it's a PFlotran mesh, and we can extract the context to get the ugrid data.

Reply:
We'll circle back to this question later.